### PR TITLE
Remove ansi decoration

### PR DESCRIPTION
**Pull request by jabaz**
Adding environment variable `DEBUG_DISABLE_COLORDECOR=1` disables ansi decoration.

### CODE REVIEW

**Comment by stephenmathieson** [comment]
isn't this the exact same thing as `DEBUG_COLORS`?

**Comment by jabaz** [comment]
No it's not, this is to be able to use colors, but disable decoration of ansi colors.
Here: https://github.com/jabaz/debug/blob/master/node.js#L103-L107
Decoration of ansi colors in this `\u001b[32;1m;` is `;1`.

Some environments doesn't support ansi color decoration.

**Comment by TooTallNate** [comment]
What does the `;1` part give us? We should probably just use the most universally compatible syntax, and not have all these obscure options.

**Comment by TooTallNate** [comment]
Oh I see, Bold.

**Comment by DevSide** [comment]
Is there a way to detect if decoration works instead ? Otherwise no bold ?

**Comment by TooTallNate** [comment]
We could perhaps take a look at the `TERM` env variable, but I doubt that will be reliable enough

**Comment by thebigredgeek** [comment]
@TooTallNate any objection to this?

@jabaz can you update docs please?

**Comment by TooTallNate** [comment]
I mean bold was nice, but we can't always have nice things 😄  We should make it a major version bump at least IMO

**Comment by jabaz** [comment]
Yes in Windows, using Vs Code @thebigredgeek
I don't know if it can be detected.

**Comment by jabaz** [comment]
It would be nice to auto detect, if supported.

**Comment by thebigredgeek** [comment]
@jabaz can you update the README, I mean?

**Comment by jabaz** [comment]
Yes I can, when i have some time.

**Comment by thebigredgeek** [comment]
@jabaz cool!  Once done I will gladly merge this :)

**Comment by TooTallNate** [comment]
> @TooTallNate any objection to this?

I'd rather not add yet anther env variable. If bold is problematic then let's just remove the bold part.

**Comment by thebigredgeek** [comment]
K

**Comment by jabaz** [comment]
Do you have another solution? Should i update README??

**Comment by thebigredgeek** [comment]
@jabaz can you simply remove bold?

**Comment by jabaz** [comment]
It's okay to me!

File: node.js [hunk 1]
```diff
@@ -53,6 +54,18 @@ function useColors() {
   }
 }
 
+function disableColorDecorations() {
+  var debugColorDecor = (process.env.DEBUG_DISABLE_COLORDECOR || '').trim().toLowerCase();
+  if (0 === debugColorDecor.length) {
+    return false
+  } else {
+    return '0' !== debugColorDecor
```
Comment (DevSide) [inline node.js]: what about `return ! /0|no|false|disabled/.test(debugColorDecor)` ?
Put the regex in a constant and use it in `useColors()` too ?

File: node.js [hunk 2]
```diff
@@ -53,6 +54,18 @@ function useColors() {
   }
 }
 
+function disableColorDecorations() {
+  var debugColorDecor = (process.env.DEBUG_DISABLE_COLORDECOR || '').trim().toLowerCase();
```
Comment (DevSide) [inline node.js]: disabled by default if var is not set and `process.platform === 'win32'` ?

**Comment by thebigredgeek** [comment]
@jabaz any update on refactoring this PR to simply remove bold?

**Comment by jabaz** [comment]
I could just set it disabled by default!

**Comment by thebigredgeek** [comment]
@jabaz Explain?  Also would like to get a solution merged in by tomorrow morning

**Comment by jabaz** [comment]
So that, if environment variable is not defined, it's disabled.
So it enables decoration if you add DEBUG_COLORDECOR = 1, yes, true or enabled.

**Comment by thebigredgeek** [comment]
@TooTallNate thoughts on above

**Comment by TooTallNate** [comment]
I don't think we need the fallback env variable (who's going to bother setting that, honestly?). Let's just remove the `1m;` and call it a day.

**Comment by jabaz** [comment]
@thebigredgeek @TooTallNate So should i just remove it?

**Comment by TooTallNate** [comment]
All the `DEBUG_COLORDECOR` and `useColorDecorations` stuff? Ya I think so.

**Comment by TooTallNate** [comment]
It'll be a v3 feature.

**Comment by jabaz** [comment]
And also remove 1m;

**Comment by TooTallNate** [comment]
Yes. Just removing the bold (and thus adding more generic compatibility) would be the new feature.

**Comment by jabaz** [comment]
@TooTallNate No it's done!

**Comment by thebigredgeek** [comment]
@jabaz merging this into v3, which will release later this year.  Trying to avoid breaking changes for now, and realized last night while talking to @TooTallNate that a lot of people are probably parsing logs with ansi characters taken into account by the parser.

**Pull request merged**